Realm/ObjectStore/schema.cpp

```diff
@@ -89,7 +89,7 @@ void Schema::validate() const
 
             // check indexable
             if (prop.is_indexed) {
-                if (prop.type != PropertyTypeString && prop.type != PropertyTypeInt) {
+                if (prop.type != PropertyTypeString && prop.type != PropertyTypeInt && prop.type != PropertyTypeBool && prop.type != PropertyTypeDate) {
```

Comment:
it'd be preferable to have an `is_indexable` bool on property type rather than inline here.